app/presenters/date_question_presenter.rb

```diff
@@ -12,22 +12,18 @@ def response_label(value)
     end
   end
 
-  def selected_day
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:day]
-  end
-
-  def selected_month
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:month]
-  end
-
-  def selected_year
-    return unless response_for_current_question.is_a? Hash
-
-    response_for_current_question[:year]
+  def parsed_response
+    return {} if current_response.blank?
+    return current_response if current_response.is_a?(Hash)
```

Comment:
Isn't the handling of `Hash` objects dealt with by `parse_input`?

Reply:
That only works if the input is valid, whereas we show the user their invalid input too.